server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,138 +290,348 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
-                    drops = Sets.difference(oldLookups.keySet(), allLookups.keySet());
-                  }
-                  if (allLookupTiers == prior_update) {
-                    LOG.debug("No updates");
-                    updateAllNewOnTier(tier, allLookups);
-                  } else {
-                    updateAllOnTier(tier, allLookups);
-                    deleteAllOnTier(tier, drops);
-                  }
-                }
-                catch (InterruptedException e) {
-                  Thread.currentThread().interrupt();
-                  throw Throwables.propagate(e);
-                }
-                catch (Exception e) {
-                  LOG.error(e, "Error updating lookups for tier [%s]. Will try again soon", tier);
-                }
+      ).get();
+
+      if (oldLookups != null) {
+        Map<String, Map<String, LookupExtractorFactoryMapContainer>> converted = new HashMap<>();
+        oldLookups.forEach(
+            (tier, oldTierLookups) -> {
+              if (oldTierLookups != null && !oldTierLookups.isEmpty()) {
+                converted.put(tier, convertTierLookups(oldTierLookups));
               }
-              prior_update = allLookupTiers;
             }
-          },
-          0,
-          lookupCoordinatorManagerConfig.getPeriod(),
-          TimeUnit.MILLISECONDS
-      );
-      Futures.addCallback(backgroundManagerFuture, new FutureCallback<Object>()
-      {
-        @Override
-        public void onSuccess(@Nullable Object result)
-        {
-          backgroundManagerExitedLatch.countDown();
-          LOG.debug("Exited background lookup manager");
+        );
+
+        configManager.set(
+            LOOKUP_CONFIG_KEY,
+            converted,
+            new AuditInfo("autoConversion", "autoConversion", "127.0.0.1")
+        );
+      }
+    }
+  }
+
+  private Map<String, LookupExtractorFactoryMapContainer> convertTierLookups(
+      Map<String, Map<String, Object>> oldTierLookups
+  )
+  {
+    Map<String, LookupExtractorFactoryMapContainer> convertedTierLookups = new HashMap<>();
+    oldTierLookups.forEach(
+        (lookup, lookupExtractorFactory) -> {
+          convertedTierLookups.put(lookup, new LookupExtractorFactoryMapContainer(null, lookupExtractorFactory));
         }
+    );
+    return convertedTierLookups;
+  }
 
-        @Override
-        public void onFailure(Throwable t)
-        {
-          backgroundManagerExitedLatch.countDown();
-          if (backgroundManagerFuture.isCancelled()) {
-            LOG.info("Background lookup manager exited");
-            LOG.trace(t, "Background lookup manager exited with throwable");
-          } else {
-            LOG.makeAlert(t, "Background lookup manager exited with error!").emit();
-          }
+  @VisibleForTesting
+  void lookupManagementLoop()
+  {
+    // Sanity check for if we are shutting down
+    if (Thread.currentThread().isInterrupted() || !lifecycleLock.awaitStarted(15, TimeUnit.SECONDS)) {
```

Comment:
`lifecycleLock.awaitStarted(..)` is reliable in case interrupt does not work for some reason, however it should.

note that start/stop are synchronized with lifecycleLock so they can't happen in parallel. 

what is wrong with separate management thread? start/stop always maintain that previous executorService is stopped completely before creating and starting a new one... and raise alerts in case that can't be maintained and stop managing lookups.